crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -72,6 +74,8 @@ impl ServerBuilder {
             port,
             host,
             schema: async_graphql::Schema::build(Query, Mutation, EmptySubscription)
+                .register_output_type::<IMoveObject>()
```

Comment:
curious: do the order in which we register them matter? I suppose if one interface depends on another

Reply:
I think it can have an effect in some cases that combine recursive types with flattening -- I was wondering if I could leverage that to get around the issues that `async-graphql` had with me using `flatten` like I had originally planned, but sadly I don't think that's possible because the damage is done in the call to `Schema::build` (it starts recursively exploring types in `Query`, `Mutation`, etc) and the circular + flattened types are left in a bad state that I don't think it's possible to recover from.